ee/clickhouse/models/property.py

```diff
@@ -21,6 +21,7 @@ def parse_prop_clauses(
     prepend: str = "global",
     table_name: str = "",
     allow_denormalized_props: bool = False,
+    filter_test_accounts=False,
```

Comment:
Hm not sure I follow. Are you saying I should just change the name of the property to something else (`should_filter_test_accounts` maybe?). I don't find it too confusing, if filter_test_accounts=True you'd expect it to filter out test accounts and vice versa.

Reply:
I meant it's not obvious when this argument should be passed vs when not. It results in one additional thing to consider in every codepath.

Given we know whether to `filter_test_accounts` or not in the filter object, a better solution (which hides this toggle from 90% of code) is to do it in the filter object in `def properties()` as proposed above.

Reply:
Hm yeah I think you are right, I though there were a few paths were we explicitly didn't want to set filter_test_accounts even if the main filter does have it.

I don't think I understand your solution, do you mind elaborating? What do you mean with overload def properties()?

Reply:
Sure.

So a call to `parse_prop_clauses` looks something like this:

```
query, params = parse_prop_clauses(filter.properties, ...)
```

The idea is to make `filter.properties` dependent on the filter_test_accounts parameter. This can be done something like this in FilterTestAccountsMixin:

```
    @cached_property
    def properties(self) -> List[Property]:
        properties = super()
        if self.filter_test_accounts: 
            properties.extend(...)
        return properties
```

Reply:
Hmm that feels pretty messy, as you won't know what you are passing through. Should we just pass through the filter object separately and then check if filter_test_accounts is true?

Reply:
That would work as well, though not sure whether all callsites have access to a filter object. :)

Reply:
Have we come full circle that the original solution is the best solution?

Reply:
I understood your proposal to be to pass `filter` object to parse_prop_clauses.

But we're going very long back-and-forth on this. If you're sold on the current approach, then let's go with that and refactor later as needed.

Reply:
I'm not sold on it but I also don't think the other proposals are much better. I think fine to leave for now, there's good tests so refactoring should be easy.